Poisson_surface_reconstruction_3/include/CGAL/Poisson_mesh_domain_3.h

```diff
@@ -0,0 +1,427 @@
+   *
+   * The method takes as argument a bounding sphere which is required to
+   * circumscribe the surface and to have its center inside the domain.
+   *
+   * \tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+   * \tparam Bounding_object either a bounding sphere (of type `Sphere_3`), a bounding box (type `Bbox_3`),
+   *                         or a bounding `Iso_cuboid_3`
+   *
+   * \param function the Poisson reconstruction function
+   * \param bounding_object object bounding the meshable domain and its center is inside the domain.
+   * \param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below:
+   *
+   * \cgalNamedParamsBegin
+   *   \cgalParamNBegin{relative_error_bound}
+   *     \cgalParamDescription{ is the relative error
+   *                            bound, relative to the diameter of the box of the image.}
+   *     \cgalParamDefault{FT(1e-3)}
+   *   \cgalParamNEnd
+   * \cgalNamedParamsEnd
+   *
+   */
+  template<typename Bounding_object, typename CGAL_NP_TEMPLATE_PARAMETERS>
+  static Poisson_mesh_domain_3 create_Poisson_mesh_domain(const Function& function,
+                                                           const Bounding_object& bounding_object,
+                                                           const CGAL_NP_CLASS& np = parameters::default_values()
+#ifndef DOXYGEN_RUNNING
+                                                           ,typename std::enable_if<!is_named_function_parameter<Function>>::type* = nullptr
+#endif
+)
+  {
+    using parameters::get_parameter;
+    using parameters::choose_parameter;
+    FT relative_error_bound_ = choose_parameter(get_parameter(np, internal_np::error_bound), FT(1e-3));
+    CGAL::Random* p_rng_ = choose_parameter(get_parameter(np, internal_np::rng), nullptr);
+    auto null_subdomain_index_ = choose_parameter(get_parameter(np, internal_np::null_subdomain_index_param),
+Null_functor()); auto construct_surface_patch_index_ = choose_parameter(get_parameter(np,
+internal_np::surface_patch_index), Null_functor());
+    namespace p = CGAL::parameters;
+    return Poisson_mesh_domain_3
+            (function,
+             bounding_object,
+              CGAL::parameters::relative_error_bound(relative_error_bound_)
+              .function(make_implicit_to_labeling_function_wrapper<BGT>(function))
+              .p_rng(p_rng_)
+              .null_subdomain_index(Base::create_null_subdomain_index(null_subdomain_index_))
+              .construct_surface_patch_index(Base::create_construct_surface_patch_index(construct_surface_patch_index_)));
+  }
+/// @}
+#ifndef DOXYGEN_RUNNING
+  template<typename CGAL_NP_TEMPLATE_PARAMETERS>
+  static Poisson_mesh_domain_3 create_Poisson_mesh_domain(const CGAL_NP_CLASS& np = parameters::default_values())
+  {
+    static_assert(!parameters::is_default_parameter<CGAL_NP_CLASS, internal_np::function_param_t>::value,
```

Comment:
What's the point of having a default parameter value in L257 if function cannot be default already. (Same in base class?)